### DISCUSSION TITLE
Shiny Apps UI

### DISCUSSION THREAD

**Discussion by naturallyfunny** [teamalgoritma/community#792] [Q&A] [answered]
# Assalamualaikum

Ada yang tau cara menampilkan video di shiny apps?

Saya sudah coba dengan cara berikut:
`tags$video(src = video.mp4", type = "video/mp4", controls = TRUE, autoplay = TRUE, width = "100%", height = "100%"),`
videonya berhasil muncul saat di-run di lokal, tapi tidak saat di-run di server shiny. Seperti ini:
<img width="500" alt="Screenshot 2023-10-31 at 7 07 03 PM" src="https://github.com/teamalgoritma/community/assets/134937547/3d985272-4560-4d10-90e7-20ac2c9f16a0">

**Comment by arkanayudhistira** [comment, marked as answer]
Dear @ArdiantheGreat ,

Jika menampilkan video menggunakan video lokal tidak dapat digunakan, dapat dicoba dengan mengupload video ke YouTube kemudian diembed dengan cara:

1. Upload video ke YouTube
2. Klik **Share** pada video YouTube yang telah terinstall 
    ![image](https://github.com/teamalgoritma/community/assets/100895120/e90ceae9-65a7-40cb-901f-3ecec823d540) 
4. Pilih **Embed** 
    ![image](https://github.com/teamalgoritma/community/assets/100895120/e1896135-fb3e-451b-a7e2-d7384e4ef670)
5. Copy code embed dari YouTube kemudian implementasikan ke Dashboard dengan fungsi `HTML()`
    ![image](https://github.com/teamalgoritma/community/assets/100895120/486fa2a4-2fc6-4aa5-86a5-13cdb6d7ae37)
6. Sehingga, hasilnya akan seperti ini:
    ```
    HTML('<iframe width="560" height="315" src="https://www.youtube.com/embed/T1-k7VYwsHg" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>')
    ```

Berikut referensi contoh implementasinya: https://community.rstudio.com/t/how-to-embed-videos-in-shiny/38937/2 

Semoga membantu!

Best Regards,
Team Algoritma
